src/data/nat/factorization.lean

```diff
@@ -119,6 +120,32 @@ begin
     simp [prod_insert hxT, sum_insert hxT, ←IH, factorization_mul (hS x hxS) hT] }
 end
 
+/-! ### Bijection between pnats and finsupps `ℕ →₀ ℕ` with support on the primes -/
+
+/-- Any finsupp `f : ℕ →₀ ℕ` whose support is in the primes is equal to the factorization of
+the product `∏ (a : ℕ) in f.support, a ^ f a`. -/
+lemma factorization_prod_pow_inv {f : ℕ →₀ ℕ} (hf : ∀ (p : ℕ), p ∈ f.support → prime p) :
+  (f.prod pow).factorization = f :=
+begin
+  have h : ∀ x : ℕ, x ∈ f.support → x ^ f x ≠ 0 := λ p hp, pow_ne_zero _ (prime.ne_zero (hf p hp)),
+  simp only [finsupp.prod, factorization_prod h],
+  nth_rewrite_rhs 0 (sum_single f).symm,
+  exact sum_congr rfl (λ p hp, prime.factorization_pow (hf p hp)),
+end
+
+lemma eq_factorization_iff {n : ℕ} {f : ℕ →₀ ℕ} (hn : n ≠ 0) (hf : ∀ p ∈ f.support, prime p) :
```

Comment:
The implications in the name and in the statement are one the opposite of the others.